lib/i18n/tasks/translators/deepl_translator.rb

```diff
@@ -60,21 +60,102 @@ def options_for_plain
 
     # @param [String] value
     # @return [String] 'hello, %{name}' => 'hello, <i18n>%{name}</i18n>'
-    def replace_interpolations(value)
+    def original_replace_interpolations(value)
       value.gsub(INTERPOLATION_KEY_RE, '<i18n>\0</i18n>')
     end
 
     # @param [String] untranslated
     # @param [String] translated
     # @return [String] 'hello, <i18n>%{name}</i18n>' => 'hello, %{name}'
-    def restore_interpolations(untranslated, translated)
+    def original_restore_interpolations(untranslated, translated)
       return translated if untranslated !~ INTERPOLATION_KEY_RE
 
       translated.gsub(%r{</?i18n>}, '')
     rescue StandardError => e
       raise_interpolation_error(untranslated, translated, e)
     end
 
+    # deepl does a better job with interpolations when it doesn't have to deal
+    # with <br> tags, so we replace all of them with meaningless asterisk chains
+    BR_REGEXP = %r{(<br\s*/?>\s*)+}i.freeze
+    BR_SINGLE_MARKER = ' *** '
+    BR_DOUBLE_MARKER = ' ***** '
+
+    # letting deepl 'read' the interpolations gives better translations (and
+    # solves the problem of interpolations getting pushed all the way to the
+    # front of the sentence), however, deepl will also try to translate
+    # the interpolations and that gets messy.
+    # we use nonsense three-letter acronyms so deepl will 'read' them and leave
+    # them alone (the letter X also works very well, except in sentences with
+    # several consecutive interpolations because that reads X X X and deepl
+    # doesn't handle that well)
+    # deepl also needs to know if an interpolation will be a word or a number,
+    # for romance languages it matters. a little Spanish lesson to illustrate:
+    # "%{foo} betalingen" translates either to "facturas %{foo}"
+    # (openstaande betalingen -> facturas pendientes) or to "%{foo} facturas"
+    # (5 betalingen -> 5 facturas)
+    # for interpolation keys that are usually numeric, we pick a number
+    # instead of the three-letter acronym (more consistency in how we name
+    # interpolation keys would help)
+    LETTER_SUBS = %w[RYX QFN VLB XOG DWP ZMQ JZQ WVS LRX HPM].freeze
+    NUM_SUBS = %w[17 19 23 29 31 37 41 43 47 53].freeze
+
+    def key_substitution(stache, index)
```

Comment:
Just an old name for inlining variables in an html string. 
Ruby calls it interpolation, which is a misuse of the mathematical term.
How about handle="" to be a bit more formal? (stache - moustache - handlebars - handle)

Reply:
'handle', or perhaps 'key', since that's what the function is called. Also in the code below, in the gzip function, you increment the 'key_index' for every 'stache' in that case I'd expect it to be called 'key'.

However, in the docs (and also in your html tag) it's called 'variables' (e.g. see https://guides.rubyonrails.org/i18n.html#passing-variables-to-translations), and the feature is called 'variable interpolation', so i'd personally opt for a signature like 'substitute_variable(variable, index)' (or 'translation_var(iable)', if 'variable' is too generic), and then also use 'variable_index' in the code below.

Reply:
@tom-brouwer-bex I'm trying, perhaps not very well, to keep a distinction between a handle that is wrapped "{{ name }}" or "%{ name }" and a key that is just "name". Just plain index looks better to me. Code updated.